x-pack/plugin/esql/compute/src/main/java/org/elasticsearch/compute/aggregation/blockhash/BlockHash.java

```diff
@@ -127,14 +132,17 @@ public static BlockHash buildPackedValuesBlockHash(List<GroupSpec> groups, Block
     /**
      * Creates a specialized hash table that maps a {@link Block} of the given input element type to ids.
      */
-    private static BlockHash newForElementType(int channel, ElementType type, BlockFactory blockFactory) {
+    private static BlockHash newForElementType(int channel, ElementType type, AggregatorMode aggregatorMode, BlockFactory blockFactory) {
```

Comment:
Hm, maybe a more natural way to pass in the info that we need a category block hash would be to alter `GroupSpec` and instead of `ElementType`, use a new class/enum that either is a group based on a simple data type or a categorization?

Reply:
Where should the logic go that indicates that the result of the `Categorize` function is a `BytesRef` that should be interpreted as a `Category` and go to the `CategoryRawBlockHash`? 

And should the no-op `Categorize` function stay or go?

Reply:
I'll need to take a closer look, but I _think_ in `AbstractPhysicalOperationProviders.groupingPhysicalOperation` we should have all required information to choose the correct block hash. Currently, this info is passed on by passing the aggregator mode and the group spec (resp. its data type) to the `HashAggregationOperatorFactory`'s constructor, but we could pass this info on by different means as well - if I'm correct.

Reply:
I was thinking we'd avoid calling these functions entirely when we know we're dealing with CATEGORIZE. But I haven't found the spot to make that choice.